### PR TITLE
Entire project review

### PR DESCRIPTION
**Pull request by pcosme**
Mock pull request.
To be able to comment code on master for review.

### CODE REVIEW

File: src/BoundaryLib.cpp [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+//#pragma omp parallel for default(none) shared(fluid_class,nx,ny)
+	for(int j=0; j < ny; j++){
+		int right;
+		right = nx - 1 + j * nx;
+		fluid_class.Den[right]=fluid_class.Den[right - 1];			//free density at x=L
+		fluid_class.FlxY[right] = 0.0f ;					//idem at x=L
+		fluid_class.FlxX[right] =  fluid_class.FlxX[right - 1];
+	}
+}
+
```
Comment (tcosme) [inline src/BoundaryLib.cpp:78]: Inconsistent spacing between functions

File: src/includes/BoundaryLib.h [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		static void YClosedNoSlip(Fluid2D& fluid_class);   ///< zero flux across y=0 and y=W and zero tangent velocity Vx=0
+};
+
+/*!
+ * @brief Class for constant Dirichelet boundary conditions
+ *
+ * The class implements boundary conditions where the variables of the system are set to a constant value along the boundary. For the fluid this can translate in a variety of situations and physical scenarios.
+ *
+ * */
+class  DirichletBoundaryCondition : public BoundaryCondition
```
Comment (tcosme) [inline src/includes/BoundaryLib.h:55]: multiple functions per file

File: src/BoundaryLib.cpp [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+//#pragma omp parallel for default(none) shared(fluid_class,nx,ny)
+	for(int j=0; j < ny; j++){
+		int left;
+		left= 0 + j * nx;
+		fluid_class.Den[left]=fluid_class.Den[left + 1];
+		fluid_class.FlxY[left] = 0.0f; 					//flux only on x at x=0
+		fluid_class.FlxX[left] = fluid_class.FlxX[left + 1] * pow(fluid_class.Den[left + 1], -1.5f);			//free flux at x=0
+	}
+}
+void BoundaryCondition::XFreeRight(Fluid2D &fluid_class) {
```
[the remark this answers is not in the document]
Comment (pcosme) [reply src/BoundaryLib.cpp:67]: I now joined the definitions of the Left and Right (as well as the Top and Bottom) into a single function with and aditional parameter to indicate the side (cf. commit 382f5596808f97f4e0642aa329ec4efe3cbf257b )

File: src/Tethys1DLib.cpp [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+float GrapheneFluid1D::VelocitySource(__attribute__((unused)) float n,float v,__attribute__((unused)) float s){
+	return -1.0f * col_freq * v ;
+}
+
+
+int Fluid1D::GetSnapshotStep() const { return snapshot_step;}
+int Fluid1D::GetSnapshotFreq() const {return snapshot_per_period;}
+
+
+bool Fluid1D::Snapshot() const {
```
[the remark this answers is not in the document]
Comment (pcosme) [reply src/Tethys1DLib.cpp:219]: Agree. Already implemented

File: src/includes/BoundaryLib.h (hunk 2 above)
Comment (pcosme) [reply src/includes/BoundaryLib.h:55]: I've already refactor and moved each class to its own file and to help the compilation  the cmake now includes all cpp via glob

**Comment by pcosme** [comment]
At commit 2d23f195dd0766a62d534bac0db971a3667878c4 
I changed the try/catch  blocks to regular returns as advised. I've left only the ones regarding the opening and writting of hdf5 files

File: src/includes/BoundaryLib.h [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+		static void XFreeLeft(Fluid2D& fluid_class);       ///< open boundaries at x=0 for all variables and zero tangent velocity Vy=0
+		static void XFreeRight(Fluid2D& fluid_class);      ///< open boundaries at x=L for all variables and zero tangent velocity Vy=0
+		static void XPeriodic(Fluid1D& fluid_class);       ///< periodic boundaries u(x=0)=u(x=L) for all variables  and zero tangent velocity
+		static void XPeriodic(Fluid2D& fluid_class);       ///< periodic boundaries u(x=0)=u(x=L) for all variables and zero tangent velocity
+		static void YFree(Fluid2D& fluid_class);           ///< open boundaries at y=0 and y=W for all variables
+		static void YFreeTop(Fluid2D& fluid_class);        ///< open boundaries at y=W for all variables
+		static void YFreeBottom(Fluid2D& fluid_class);     ///< open boundaries at y=0 for all variables
+		static void YPeriodic(Fluid2D& fluid_class);       ///< periodic boundaries u(y=0)=u(y=W) for all variables
+		static void YClosedFreeSlip(Fluid2D& fluid_class); ///< zero flux across y=0 and y=W and free tangent velocity Vx
+		static void YClosedNoSlip(Fluid2D& fluid_class);   ///< zero flux across y=0 and y=W and zero tangent velocity Vx=0
```
[the remark this answers is not in the document]
Comment (pcosme) [reply src/includes/BoundaryLib.h:46]: I'd rather have them in separate classes that will allow not only easier reading of the code but also will became easier to derive new kinds of boundary conditions (i.e. new derived classes )

File: src/BoundaryLib.cpp [hunk 6]
```diff
@@ -0,0 +1,518 @@
...
+		fluid_class.FlxY[top] = 0.0f;
+	}
+}
+
+void BoundaryCondition::SetBottomEdge(Fluid2D &fluid_class) {
+	int nx=fluid_class.SizeX();
+	int ny=fluid_class.SizeY();
+	BottomEdge = new int[nx]();
+	for(int i=0;i<nx;i++){
+		for(int j=0;j<ny/2;j++) {
```
[the remark this answers is not in the document]
Comment (pcosme) [reply src/BoundaryLib.cpp:222]: The implementation of non-rectangular domain (cf. issue #50 ) is still at a very early stage and will probably be redone entirely and not only the nested cycles

File: src/ElectricLib.cpp [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+	transform(DipVarVarX.begin(), DipVarVarX.end(), DipVarVarX.begin(), [dt](const float &c){ return c/(dt*dt); });
+	transform(DipVarY.begin(), DipVarY.end(), DipVarY.begin(), [dt](const float &c){ return c/dt; });
+	transform(DipVarVarY.begin(), DipVarVarY.end(), DipVarVarY.begin(), [dt](const float &c){ return c/(dt*dt); });
+}
+
+
+void ElectroAnalysis::WriteElectroFile() {
+	try{
+		if(NetQ.empty() || DipVarY.empty()){
+			throw "Nothing to save on output file";
```
[the remark this answers is not in the document]
Comment (pcosme) [reply src/ElectricLib.cpp:81]: check commit 2d23f195dd0766a62d534bac0db971a3667878c4

File: src/Tethys1DLib.cpp [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+float GrapheneFluid1D::DensityFlux(float n,float v,float __attribute__((unused)) s){
+	float f_1;
+	f_1 = n * v;
+	return f_1;
+}
+float GrapheneFluid1D::VelocityFlux(float n,float v,__attribute__((unused))float dv,float s){
+	float f_2;
+	try {
+		if(n<0 || !isfinite(n)){
+			throw "ERROR: negative density";
```
[the remark this answers is not in the document]
Comment (pcosme) [reply src/Tethys1DLib.cpp:198]: check commit 2d23f195dd0766a62d534bac0db971a3667878c4

File: src/includes/TethysLib.h [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+		float ImagFreq() const;
+		float PhaseVel() const;
+		float RealFreq() const;
+
+
+		std::string GetInfix() const;   ///< @returns   file name infix
+
+
+		void CreateHdf5File();          ///< creates the HDF5 files with the necessary structure
```
[the remark this answers is not in the document]
Comment (pcosme) [reply src/includes/TethysLib.h:153]: I agree that it can be an improvement although probably not of first priority

**Pull request closed without merging**